backend/data_distro/management/commands/validate_api.py

```diff
@@ -0,0 +1,88 @@
+
+"""
+Download data from https://facdissem.census.gov/PublicDataDownloads.aspx
+Then unzip the files and place the them in data_distro/data_to_load/
+
+Load them with: manage.py public_data_loader
+"""
+import json 
+from django.core.management.base import BaseCommand
+
+from data_distro.api_tests import (
+    fac,
+    schemas
+)
+from data_distro.management.commands.handle_errors import (
+    set_up_error_files,
+    log_results,
+)
+
+
+class Command(BaseCommand):
+    help = """
+        Runs validations on the API endpoints.
+
+        Becase the API presents *views* of the data, these tests run JSON Schema validations
+        over the data when round-tripped from the Census import through to the PostgREST 
+        API export.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument("--view", type=str, help="API view", default="vw_general")
+        parser.add_argument(
+            "-y",
+            "--year",
+            type=int,
+            help="Four digit year to be validated",
+            default=2022
+            )
+        parser.add_argument("--start", type=int, help="Result set start location", default=0)
+        parser.add_argument("--end", type=int, help="Result end location", default=100)
+        parser.add_argument("--step", type=int, help="Result set step size", default=None)
+
+    def handle(self, *args, **kwargs):
+        print(kwargs)
+        if kwargs["step"] == None:
+            kwargs["step"] = (kwargs["end"] - kwargs["start"]) // 2
+         
+        failure_count = 0
+        validation_errors = []
+        result_set = (fac.FAC()
+                        .table(kwargs["view"])
+                        .limits(kwargs["start"], kwargs["end"], kwargs["step"])
+                        #.query("eq", "audit_year", kwargs["year"])
+                        ).run()
+        if len(result_set) == 0:
+            print(f"No results: {result_set}")
+
+        for obj in result_set:
+            validation_result = schemas.validate_api_object(kwargs["view"], obj)
+            
+            item_errors = []
+            for e in validation_result:
+                    item_errors.append(e)
+                    failure_count += 1
+
+            if any(item_errors):
+                validation_errors.append({"object": obj, "error": item_errors})
+
+        for oe in validation_errors:
+            oe_id = oe["object"]["id"]
+            messages = ""
+            for k, error in enumerate(oe["error"]):
+                messages += f"{k+1}. {error}\n\n"
+            
+            content = """
+-------------
+OBJECT id: {oe_id}
+-------------
+{oe_object}
+
+------
+ERRORS
+------
+{messages}
+"""
+
+            print(content.format( oe_id=oe_id, oe_object=oe["object"], messages=messages))
+        return f"failure_count: {failure_count}"
```

Comment:
I made a bunch of changes above so we could print multiple validation errors for a given object. Please excuse and/or issue stern corrections for any poor python etiquette.

Reply:
That `validate_api` script was written by a non-pythonista, so... :shrug:. For now, it's a good thing that we're doing it.